homeassistant/setup.py

```diff
@@ -240,19 +242,39 @@ def log_error(msg: str) -> None:
     elif name in processed:
         return
 
-    if hasattr(module, 'DEPENDENCIES'):
+    manifest = await hass.async_add_executor_job(
+        json_util.load_json,
+        os.path.join(os.path.dirname(module.__file__), 'manifest.json')
+    )
+
+    if manifest.get('dependencies'):
+        dep_success = await _async_process_dependencies(
```

Comment:
Instead of duplicating this code, let's extract the "find dependencies" part above this method. That way we're not duplicating the call to `_async_process_dependencies`. Same for requirements.